tests/test_signatory_component.cairo

```diff
@@ -0,0 +1,195 @@
+    start_cheat_caller_address(contract_address, token_owner);
+    let is_valid_signer = signatory_dispatcher.is_valid_signer(token_owner);
+
+    assert(is_valid_signer == true, 'should accept valid signature');
+    stop_cheat_caller_address(contract_address);
+}
+
+#[test]
+fn test_is_valid_signer_permissioned_address() {
+    let (contract_address, erc721_contract_address) = __setup__();
+
+    let token_dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    let token_owner = token_dispatcher.ownerOf(1.try_into().unwrap());
+    let permissionable_dispatcher = IPermissionableDispatcher { contract_address };
+    let signatory_dispatcher = ISignatoryDispatcher { contract_address: contract_address };
+
+    let mut permission_addresses = ArrayTrait::new();
+    permission_addresses.append(ACCOUNT2.try_into().unwrap());
+    permission_addresses.append(ACCOUNT3.try_into().unwrap());
+
+    let mut permissions = ArrayTrait::new();
+    permissions.append(true);
+    permissions.append(true);
+
+    start_cheat_caller_address(contract_address, token_owner);
+    permissionable_dispatcher.set_permission(permission_addresses, permissions);
+    let is_valid_signer = signatory_dispatcher.is_valid_signer(ACCOUNT2.try_into().unwrap());
+
+    assert(is_valid_signer == true, 'should accept valid signer');
+    stop_cheat_caller_address(contract_address);
+}
+
+#[test]
+fn test_is_valid_signer_root_owner() {
+    let (contract_address, erc721_contract_address) = __setup__();
+
+    let token_dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    let token_owner1 = token_dispatcher.ownerOf(1.try_into().unwrap());
+    token_dispatcher.ownerOf(2.try_into().unwrap());
+
+    let signatory_dispatcher = ISignatoryDispatcher { contract_address: contract_address };
+
+    start_cheat_caller_address(contract_address, token_owner1);
+
+    let is_valid_signer = signatory_dispatcher.is_valid_signer(token_owner1);
+
+    assert(is_valid_signer == true, 'should accept valid signer');
+    stop_cheat_caller_address(contract_address);
+}
+
+
+#[test]
+fn test_is_valid_signature() {
+    let (contract_address, erc721_contract_address) = __setup__();
+
+    let data = SIGNED_TX_DATA();
+    let hash = data.transaction_hash;
+
+    let token_dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    let token_owner = token_dispatcher.ownerOf(1.try_into().unwrap());
+    let signatory_dispatcher = ISignatoryDispatcher { contract_address: contract_address };
+
+    start_cheat_caller_address(contract_address, token_owner);
+    let mut good_signature = array![data.r, data.s];
+    let is_valid = signatory_dispatcher.is_valid_signature(hash, good_signature.span());
+    assert(is_valid == 'VALID', 'should accept valid signature');
+    stop_cheat_caller_address(contract_address);
+}
+
+#[test]
+fn test_is_valid_signature_root_owner() {
```

Comment:
wrong test implementation! you are meant to test for the root owner not the token owner! get the root owner by calling the `_get_root_owner` function on the tokenbound account and testing against that instead.

Hint: you might want to expose the `_get_root_owner` by adding it as an extra function to the preset account.